internal/services/apicenter/api_center_environment_resource.go

```diff
@@ -0,0 +1,304 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+package apicenter
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/apicenter/2024-03-01/environments"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/apicenter/2024-03-01/services"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+type ApiCenterEnvironmentResource struct{}
+
+var _ sdk.ResourceWithUpdate = ApiCenterEnvironmentResource{}
+
+type ApiCenterEnvironmentResourceModel struct {
+	Name           string `tfschema:"name"`
+	ServiceId      string `tfschema:"service_id"`
+	Identification string `tfschema:"identification"`
+	Description    string `tfschema:"description"`
+	Type           string `tfschema:"environment_type"`
+	DevPortalUri   string `tfschema:"development_portal_uri"`
+	Instructions   string `tfschema:"instructions"`
+	ServerType     string `tfschema:"server_type"`
+	MgmtPortalUri  string `tfschema:"management_portal_uri"`
+}
+
+func (r ApiCenterEnvironmentResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+		},
+
+		"service_id": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+		},
```

Comment:
Can we rename this to be more explicit and also the ID validator on it
```suggestion
		"api_center_service_id": {
			Type:     pluginsdk.TypeString,
			Required: true,
			ForceNew: true,
		},
```